GoogleForCreators/web-stories-wp
# Editor: Animations: Add initial play functionality to editor

## Summary
- Updates pagemenu icons to be closer in size to new designs https://www.figma.com/file/Cd9iacVknHK62RO4pRwb68/Stories-Editor?node-id=11070%3A16
- Incorporates `StoryAnimation.Provider` into editor
- Adds play button to play anims
- Stops animation if page changes or design panel focused
- Makes elements unselectable if animation is playing
- Removes movable controls, hover highlights & pointer events from frame elements while animations playing or scrubbing
- Moves animation control behind `enableAnimation` flag in editor

## Relevant Technical Choices
Renamed animations states object constant to `STORY_ANIMATION_STATE` since it made more sense to me.

Moved the animation provider in with the display layer. Figured it was part of the display layer so should live there. also needed to be downstream of the story provider where I put the animation state.

## To-do
NA

## User-facing changes
Page menu icons made smaller to fit updated designs.

## Testing Instructions
set `enableAnimation` to `true` in `Story_Post_Type.php` and open a story created from a template with animations in the editor. See that there's now a play button on the bottom right side.

- Press the play button and see the animations play.
- Select an element then press play and make sure movable controls dissappear when animation playing and reappear when animation complete.
- Select an element then press play, while animation playing, select a field in the style editor. see that animation promptly stops.
- Press play button. while page is animating press the next page button. Next page should render stopped with play button able to press.
- Press play button. while page is animating, try to select an element on the page. Nothing should happen as elements should be unselectable while page is animating.

Select a story you've created that has no animations on it. In editor, make sure play button is there but disabled.

## Screenshots
1. Selecting anything in design panel while animation is playing resets animation to initial stopped state.
![stop_animation_design_panel](https://user-images.githubusercontent.com/35983235/89062513-d50df800-d323-11ea-81bd-c76b092278f7.gif)

2. Movable boxes disappear & reappear while animation is playing and when it finished
![selection_functionality](https://user-images.githubusercontent.com/35983235/89062522-d7705200-d323-11ea-953f-e0bee25456bb.gif)

3. Animation stops when page changed & new page is stopped & playable
![play_functionality](https://user-images.githubusercontent.com/35983235/89062529-d93a1580-d323-11ea-9824-3c323fdd4eb7.gif)

4. (with animation flag disabled) Icon size is closer to new designs
<img width="1440" alt="Screen Shot 2020-07-31 at 11 03 28 AM" src="https://user-images.githubusercontent.com/35983235/89062534-db03d900-d323-11ea-80ab-8540a690d778.png">

## QA Check:
Had to make small changes to the buttons in the dock right beneath the page view in the editor so I could fit the play button in here. I believe these have since been updated again with the latest design pass. The test is just to make sure there's no regressions here with sizing/spacing of these button

<img width="416" alt="Screen Shot 2020-08-10 at 12 11 19 PM" src="https://user-images.githubusercontent.com/35983235/89815863-bcab9380-db02-11ea-9a87-a4914223f653.png">


---



Fixes #3484
